python/lsst/ts/mtmount/mtmount_csc.py

```diff
@@ -986,7 +1027,35 @@ async def _disable_devices_impl(self):
 
     async def handle_summary_state(self):
         if self.summary_state == salobj.State.FAULT:
-            await self.disable_devices()
+            try:
+                await self.disable_devices()
+            finally:
+                await self.log_command_history()
+        self.command_history.clear()
+        self.command_reply_history.clear()
+
+    async def log_command_history(self):
+        """Log the command history."""
+        self.log.info("Logging command history.")
+        try:
+            command_history = (
+                f"Command history contains {len(self.command_history)} commands. "
+                f"Response history containd {len(self.command_reply_history)}.\n\n"
+            )
+            if len(self.command_history) > 0:
+                command_history += "Command history:\n\n"
+                for command in self.command_history:
+                    command_history += f"{command}\n"
+
+            if len(self.command_reply_history):
+                command_history += "Reply history:\n\n"
+
+                for reply in self.command_reply_history:
+                    command_history += f"{reply}\n"
+
+            self.log.error(command_history)
+        except Exception:
```

Comment:
Which Exception can possibly be raised in the contained block? Maybe add an inline comment to clarify that?

Reply:
to be honest, I don't know, I just want to make sure this never errors out.